src/test/kotlin/com/example/passmanager/repositories/PassOwnerRepositoryImplTest.kt

```diff
@@ -31,70 +30,70 @@ class PassOwnerRepositoryImplTest {
     fun `finding pass by existing id should return pass owner by id`() {
         // GIVEN
         val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
 
         // WHEN
-        val passOwnerById = passOwnerRepository.findById(insertedPassOwnerId.toString())
+        val passOwnerById = insertedPassOwner
+            .mapNotNull { it.id.toString() }
+            .flatMap { insertedPassOwnerId -> passOwnerRepository.findById(insertedPassOwnerId) }
 
         // THEN
-        assertThat(passOwnerById?.id).isNotNull().isEqualTo(insertedPassOwnerId)
+        passOwnerById.test()
+            .assertNext { assertNotNull(it) }
+            .verifyComplete()
     }
 
     @Test
     fun `inserting pass owner in collection should return created pass owner`() {
         // WHEN
         val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
 
         // THEN
-        assertThat(mongoTemplate.findById<MongoPassOwner>(insertedPassOwnerId!!)).isEqualTo(insertedPassOwner)
+        insertedPassOwner
+            .mapNotNull { it.id }
+            .test()
+            .assertNext { ownerId ->
+                mongoTemplate.findById<MongoPassOwner>(ownerId!!)
+                    .mapNotNull { it.id }
+                    .doOnNext { id -> assertThat(ownerId).isEqualTo(id) }
+                    .subscribe()
+            }
```

Comment:
You have assertNext, but it doesn't assert anything
Also, why are we using mongoTemplate instead of our repository?

Reply:
1. Actually it asserts entity equality (I refactored test to compare full objects, and not only ids)
2. It is repository method tests, that is why while testing I can rely only on mongo template, not on other repository methods